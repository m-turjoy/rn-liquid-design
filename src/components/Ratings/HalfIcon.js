import React from 'react'
import { string } from 'prop-types'
import { IconWrapper, IconsWrapper } from './styled'
import Icon from '../MerckIcons'
import colors from '../../config/colors'

const HalfIcon = ({ type, color }) => (
  <IconsWrapper>
    <IconWrapper
      position='absolute'
    >
      <Icon
        name={type}
        size={24}
        color={colors.sensitiveGreyDarkest}
      />
    </IconWrapper>
    <IconWrapper
      backgroundColor={colors.transparent}
    >
      <Icon
        name={type === 'dot' ? 'halfDot' : 'halfStar'}
        size={24}
        color={color}
      />
    </IconWrapper>
  </IconsWrapper>
)

HalfIcon.propTypes = {
  type: string,
  color: string
}
HalfIcon.defaultProps = {
  type: 'star'
}

export default HalfIcon
