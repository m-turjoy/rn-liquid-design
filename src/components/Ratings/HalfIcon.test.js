import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import HalfIcon from './HalfIcon'

configure({ adapter: new Adapter() })
describe('HalfIcon component', () => {
  describe('Renders correctly', () => {
    test('it renders Default star HalfIcon', () => {
      const tree = shallow(
        <HalfIcon />,
      )
      expect(tree).toMatchSnapshot()
    })
    test('it renders Dot HalfIcon', () => {
      const tree = shallow(
        <HalfIcon
          type='dot'
        />,
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
