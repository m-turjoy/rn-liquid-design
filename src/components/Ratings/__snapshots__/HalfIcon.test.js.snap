// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HalfIcon component Renders correctly it renders Default star HalfIcon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HalfIcon
    type="star"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(View)
          position="absolute"
        >
          <Icon
            color="#e9e9ed"
            name="star"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
        <Styled(View)
          backgroundColor="transparent"
        >
          <Icon
            name="halfStar"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            color="#e9e9ed"
            name="star"
            size={24}
            themeName="vibrantCyan"
          />,
          "position": "absolute",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#e9e9ed",
            "name": "star",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "transparent",
          "children": <Icon
            name="halfStar"
            size={24}
            themeName="vibrantCyan"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": undefined,
            "name": "halfStar",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(View)
            position="absolute"
          >
            <Icon
              color="#e9e9ed"
              name="star"
              size={24}
              themeName="vibrantCyan"
            />
          </Styled(View)>,
          <Styled(View)
            backgroundColor="transparent"
          >
            <Icon
              name="halfStar"
              size={24}
              themeName="vibrantCyan"
            />
          </Styled(View)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              color="#e9e9ed"
              name="star"
              size={24}
              themeName="vibrantCyan"
            />,
            "position": "absolute",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#e9e9ed",
              "name": "star",
              "size": 24,
              "themeName": "vibrantCyan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "transparent",
            "children": <Icon
              name="halfStar"
              size={24}
              themeName="vibrantCyan"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": undefined,
              "name": "halfStar",
              "size": 24,
              "themeName": "vibrantCyan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`HalfIcon component Renders correctly it renders Dot HalfIcon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HalfIcon
    type="dot"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(View)
          position="absolute"
        >
          <Icon
            color="#e9e9ed"
            name="dot"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
        <Styled(View)
          backgroundColor="transparent"
        >
          <Icon
            name="halfDot"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            color="#e9e9ed"
            name="dot"
            size={24}
            themeName="vibrantCyan"
          />,
          "position": "absolute",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#e9e9ed",
            "name": "dot",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "transparent",
          "children": <Icon
            name="halfDot"
            size={24}
            themeName="vibrantCyan"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": undefined,
            "name": "halfDot",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(View)
            position="absolute"
          >
            <Icon
              color="#e9e9ed"
              name="dot"
              size={24}
              themeName="vibrantCyan"
            />
          </Styled(View)>,
          <Styled(View)
            backgroundColor="transparent"
          >
            <Icon
              name="halfDot"
              size={24}
              themeName="vibrantCyan"
            />
          </Styled(View)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              color="#e9e9ed"
              name="dot"
              size={24}
              themeName="vibrantCyan"
            />,
            "position": "absolute",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#e9e9ed",
              "name": "dot",
              "size": 24,
              "themeName": "vibrantCyan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "transparent",
            "children": <Icon
              name="halfDot"
              size={24}
              themeName="vibrantCyan"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": undefined,
              "name": "halfDot",
              "size": 24,
              "themeName": "vibrantCyan",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
